mlflow/server/handlers.py

```diff
@@ -1360,20 +1360,34 @@ def _validate_non_local_source_contains_relative_paths(source: str):
     "/models/artifacts/../../../"
     "s3:/my_bucket/models/path/../../other/path"
     "file://path/to/../../../../some/where/you/should/not/be"
+    "mlflow-artifacts://host:port/..%2f..%2f..%2f..%2f"
+    "http://host:port/api/2.0/mlflow-artifacts/artifacts%00"
     """
+    invalid_source_error_message = (
+        f"Invalid model version source: '{source}'. If supplying a source as an http, https,"
+        "local file path, ftp, objectstore, or mlflow-artifacts uri, an absolute path must be"
+        "provided without relative path references present."
+        "Please provide an absolute path."
+    )
+
+    while urllib.parse.unquote_plus(source) != source:
+        source = urllib.parse.unquote_plus(source)
```

Comment:
```suggestion
    while (unquoted := urllib.parse.unquote_plus(source)) != source:
        source = unquoted
```

ChatGPT told me this magic. I actually not a big fan of this code, we can avoid calling `urllib.parse.unquote_plus`.